src/utils.rs

```diff
@@ -57,14 +59,49 @@ pub fn increment_le(x: &mut [u8]) {
 pub fn add_le(x: &mut [u8], y: &[u8]) -> Result<(), ()> {
     if x.len() == y.len() {
         unsafe {
-            ffi::sodium_add(x.as_mut_ptr(), y.as_ptr(), x.len());
+                ffi::sodium_add(x.as_mut_ptr(), y.as_ptr(), x.len());
         }
         Ok(())
     } else  {
         Err(())
     }
 }
 
+/// `bin2hex()` takes raw bytes as `bin`, converts it to hexadecimal and puts the result in `hex`.
+/// `hex` contains a nul byte (\0) terminator
+/// `bin2hex()` will return Err<()> if length of `hex` is not sufficient to hold the hex representation of `bin`.
+/// Refer docs at https://download.libsodium.org/doc/helpers#hexadecimal-encoding-decoding
+pub fn bin2hex(hex: &mut [u8], bin: &[u8]) -> Result<(), ()> {
+    if hex.len() <= 2*bin.len() {
+        return Err(());
+    }
+    unsafe {
+        ffi::sodium_bin2hex(hex.as_mut_ptr() as *mut _, 2*bin.len()+1, bin.as_ptr(), bin.len());
+    }
+    Ok(())
+}
+
+/// `hex2bin()` takes bytes of a hexadecimal string in `hex`, converts it to raw bytes and puts the result in `bin`
+/// `ignore` is an optional parameter, taking a byte representation of a string to skip
+/// `bin_len` is the length of raw bytes that `hex` was converted to.
```

Comment:
converted from

Reply:
From? The hex is being converted to. Btw, the docs, don't say anything about it.